functions/node/src/index.ts

```diff
@@ -0,0 +1,25 @@
+import { initializeApp } from 'firebase-admin/app';
+import { getFirestore } from 'firebase-admin/firestore';
+import { beforeUserCreated, beforeUserSignedIn, HttpsError, type AuthBlockingEvent } from 'firebase-functions/identity';
+
+initializeApp();
+
+const db = getFirestore();
+
+async function checkUser(event: AuthBlockingEvent) {
+  const email = event.data?.email;
+
+  if (!email) {
+    throw new HttpsError('invalid-argument', 'Email is required');
+  }
+
+  const ref = db.collection('authorized-users').doc(email);
+  const doc = await ref.get();
+
+  if (!doc.exists) {
+    throw new HttpsError('not-found', `User (${email}) is not authorized`);
```

Comment:
I suggest changing this to an unauthorized (401) error instead a 404

Reply:
`unauthorized` is not a [valid code](https://firebase.google.com/docs/reference/node/firebase.functions#functionserrorcode). Maybe `permission-denied`?

Reply:
I think 'unauthenticated' is what I was after: The request does not have valid authentication credentials for the operation.